docs: Add wait for instance termination to changelog (#6749)
aws/karpenter-provider-aws: website/content/en/preview/upgrading/upgrade-guide.md

  * Expiration is now forceful and begins draining as soon as it’s expired. Karpenter does not wait for replacement capacity to be available before draining, but will start provisioning a replacement as soon as the node is expired and begins draining.
  * Karpenter's generated NodeConfig now takes precedence when generating UserData with the AL2023 `amiFamily`. If you're setting any values managed by Karpenter in your AL2023 UserData, configure these through Karpenter natively (e.g. kubelet configuration fields).
  * Karpenter now adds a `karpenter.sh/unregistered:NoExecute` taint to nodes in injected UserData when using alias in AMISelectorTerms or non-Custom AMIFamily. When using `amiFamily: Custom`, users will need to add this taint into their UserData, where Karpenter will automatically remove it when provisioning nodes.

>   * Karpenter now waits for underlying instances to be completely terminated before removing the associated nodes and nodeclaims. This implies that it may now take longer for nodes to be deleted and for nodeclaims to get cleaned up.

Reviewer: ```suggestion
  * Karpenter now waits for underlying instances to be completely terminated before removing the associated nodes. This means nodes may take longer to be deleted and for nodeclaims to get cleaned up.
```
